pulpcore/tasking/tasks.py

```diff
@@ -308,6 +314,10 @@ def dispatch(func, resources, args=None, kwargs=None, task_group=None):
                 "3.16 along with the `USE_NEW_WORKER_TYPE` setting."
             )
         )
+        # There is only exclusive use in the legacy tasking system
+        if shared_resources:
+            resources = resources[:]
+            resources.extend(shared_resources)
```

Comment:
Maybe that is more readable:
```suggestion
            resources = resources + shared_resources
```